openhgnn/models/MAGNN.py

```diff
@@ -161,28 +199,27 @@ def forward(self, g, feat_dict=None):
         """
         if feat_dict is None:
             feat_dict = g.ndata['feat']
-        with g.local_scope():
 
-            # input projection
-            for ntype in self.input_projection.keys():
-                g.nodes[ntype].data['feat'] = self.feat_drop(self.input_projection[ntype](feat_dict[ntype]))
+        h = {}
+        for ntype in self.input_projection.keys():
+            h[ntype] = self.feat_drop(self.input_projection[ntype](feat_dict[ntype]))
 
-            # hidden layer
-            for i in range(self.num_layers - 1):
-                h, _ = self.layers[i](g, self.metapath_idx_dict)
-                for key in h.keys():
-                    h[key] = self.activation(h[key])
-                g.ndata['feat'] = h
+        # hidden layer
+        for i in range(self.num_layers - 1):
+            h, _ = self.layers[i](h, self.metapath_idx_dict)
+            for ntype in h.keys():
+                h[ntype] = self.activation(h[ntype])
 
-            # output layer
-            h_output, embedding = self.layers[-1](g, self.metapath_idx_dict)
+        # output layer
+        h_output, embedding = self.layers[-1](h, self.metapath_idx_dict)
 
-            return h_output
+        # TODO: TEST!!!NEED TO DEPRECATE THE RETURN EMBEDDING!
```

Comment:
You need to write why you are returning the embedding (for now), and also why you actually don't want it.